aten/src/TH/generic/THTensor.cpp

```diff
@@ -351,12 +351,17 @@ void THTensor_(select)(THTensor *self, THTensor *src, int dimension, int64_t sli
 
   THTensor_(set)(self, src);
   THTensor_(narrow)(self, NULL, dimension, sliceIndex, 1);
+
+  std::vector<int64_t> newSize(self->dim()-1);
+  std::vector<int64_t> newStride(self->dim()-1);
```

Comment:
You need to explicitly initialize everything `< dimension` now, this is missing.

Reply:
pushed the new changes